api/demo/init.go

```diff
@@ -0,0 +1,66 @@
+package demo
+
+import (
+	"github.com/pkg/errors"
+	portainer "github.com/portainer/portainer/api"
+	"github.com/portainer/portainer/api/dataservices"
+)
+
+func initDemoUser(
+	store dataservices.DataStore,
+	cryptoService portainer.CryptoService,
+) error {
+
+	password, err := cryptoService.Hash("tryportainer")
+	if err != nil {
+		return errors.WithMessage(err, "failed creating password hash")
+	}
+
+	admin := &portainer.User{
+		Username: "admin",
+		Password: password,
+		Role:     portainer.AdministratorRole,
+	}
+
+	err = store.User().Create(admin)
+	return errors.WithMessage(err, "failed creating user")
+}
+
+func initDemoLocalEndpoint(store dataservices.DataStore) error {
+	localEndpoint := &portainer.Endpoint{
+		ID:        portainer.EndpointID(1),
```

Comment:
Unfortunately, `store.Endpoint().Create()` doesn't set a next id (don't know why, but hopefuly we could change it in future), but we could call `store.Endpoint().GetNextIdentifier()` to get the id and assign it to `ID` here.
Method could return the same id back.

Reply:
done